server/src/main/java/com/paypal/selion/grid/servlets/GridAutoUpgradeDelegateServlet.java

```diff
@@ -114,21 +117,34 @@ protected void process(HttpServletRequest request, HttpServletResponse response)
             if (idList == null) {
                 // there were no nodes that failed to auto upgrade
                 for (RemoteProxy eachProxy : this.getRegistry().getAllProxies()) {
-                    SeLionRemoteProxy proxy = (SeLionRemoteProxy) eachProxy;
-                    if (!proxy.release(downloadJSON)) {
-                        pendingProxy.add(proxy.getId());
+                    // TODO :: Address the assumption here that any node which uses SeLionRemoteProxy also has
+                    // NodeAutoUpgradeServlet available. Without truth in this assumption, the call to
+                    // proxy.release() will do nothing.
+                    try {
+                        SeLionRemoteProxy proxy = (SeLionRemoteProxy) eachProxy;
+                        if (!proxy.release(downloadJSON)) {
+                            pendingProxy.add(proxy.getId());
+                        }
+                    } catch (ClassCastException e) {
```

Comment:
@mach6 Instead of catching ClassCastException, We can add if condition to verify is this instance of SeLionRemoteProxy and execute the code. Else condition log the information for user about "Node incompatible for AutoUpgrade"